cli/cmd/service/set_role.go

```diff
@@ -0,0 +1,71 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+)
+
+func SetRoleCmd(ch *cmdutil.Helper) *cobra.Command {
+	var name, projectName, role string
+	setRoleCmd := &cobra.Command{
+		Use:   "set-role",
+		Short: "Set role for service",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var roleOptions []string
+			if projectName != "" {
+				roleOptions = projectRoles
+			} else {
+				roleOptions = orgRoles
+			}
+
+			err := cmdutil.StringPromptIfEmpty(&name, "Enter service name")
+			if err != nil {
+				return err
+			}
+			err = cmdutil.SelectPromptIfEmpty(&role, "Select role", roleOptions, "")
+			if err != nil {
+				return err
+			}
```

Comment:
Should check `ch.Interactive`. Although I also think it's fine not to have prompting in `set-role` since it's kind of rare to use it (prompts are mainly important for common commands)